src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -442,11 +442,10 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                 }
                                 UpdateTimeSeriesNameIfNeeded(context, docId, segment, tss);
                                 
-                                var segmentChangeVector = context.GetChangeVector(segment.ChangeVector);
-                                if (tss.TryAppendEntireSegment(context, segment, docId, segment.Name, baseline))
+                                if (tss.TryAppendEntireSegment(context, segment, changeVectorVersion, docId, segment.Name, baseline))
```

Comment:
I think we need to put the segment with its original change vector here, but when we compare the two segments we should use the version part

Reply:
done